### PR TITLE
Feature: follow redirects

### PR DESCRIPTION
**Pull request by Beanow**
Adds two options to a request. `followRedirects` and `maxRedirects`. Based mainly on how cURL supports it.

### CODE REVIEW

File: src/RequestData.php [hunk 1]
```diff
@@ -78,4 +90,34 @@ public function __toString()
...
+            case 302:
+                $this->method = 'GET';
+
+            //Note: 303, 307, 308 status is not supported in HTTP/1.0.
+            // case 303:
+            // case 307:
+            // case 308:
+
+                //Of course switch the location.
+                $this->url = $location;
```
Comment (rbraband) [inline src/RequestData.php]: to refer relativ redirect locations we need a test and url rebuild.
```PHP
//Of course switch the location.
if (is_null(parse_url($location, PHP_URL_HOST))) {
    $path = parse_url($location, PHP_URL_PATH) ?: '/';
    $queryString = parse_url($location, PHP_URL_QUERY);
    $location = $this->getScheme() . '://' . $this->getHost() . $path . ($queryString ? "?$queryString" : '');
}
$this->url = $location;
```

**Comment by Beanow** [comment]
I gave edit rights for my branch to the maintainers.
Unfortunately I won't have time to apply any feedback myself.

File: src/RequestData.php (hunk 1 above)
Comment (WyriHaximus) [inline src/RequestData.php]: Another way to do that is to use PSR7 and resolve two URI's with https://github.com/guzzle/psr7/blob/master/src/UriResolver.php

**Comment by clue** [comment]
As much as I'd love to get this feature in, I'm having to close this for now as it hasn't received any input in a while and it's unlikely this will get traction any time soon.

The feature request is still open in #40 and we'll look into this in the not too far future :+1: I believe that getting this in is also far more reasonable once the PSR-7 support is in via #41. If you feel this was closed prematurely or want to pick this up again, please let us know and we can reopen this.

Thank you for your effort!

**Pull request closed without merging**